app/src/main/java/io/appium/uiautomator2/utils/AXWindowHelpers.java

```diff
@@ -99,20 +106,49 @@ private static AccessibilityNodeInfo[] getWindowRoots() {
         return result.toArray(new AccessibilityNodeInfo[0]);
     }
 
+    private static AccessibilityNodeInfo getTopmostWindowRootFromActivePackage() {
+        CharSequence activeRootPackageName = checkNotNull(getActiveWindowRoot().getPackageName());
+
+        List<AccessibilityWindowInfo> windows = getWindows();
+        Collections.sort(windows, new Comparator<AccessibilityWindowInfo>() {
+            @Override
+            public int compare(AccessibilityWindowInfo w1, AccessibilityWindowInfo w2) {
+                return Integer.compare(w2.getLayer(), w1.getLayer()); // descending order
+            }
+        });
+        for (AccessibilityWindowInfo window : windows) {
+            AccessibilityNodeInfo root = window.getRoot();
+            if (root != null &&
+                    Objects.equals(root.getPackageName(), activeRootPackageName)) {
+                return root;
+            }
+        }
+        throw new UiAutomator2Exception("No active window root found");
```

Comment:
We could improve the error message like 

Cannot find the active topmost window belonging to xxx package

Reply:
I agree that it is understandable when the package name is shown. Revised the error message with 9cff0a99d9afbc96bff7b7b4e40dbd560fc594cf.

Thank you.